components/event-receiver/org.wso2.carbon.event.receiver.core/src/main/java/org/wso2/carbon/event/receiver/core/internal/EventReceiver.java

```diff
@@ -239,7 +248,8 @@ protected void sendEvent(Event event) {
             statisticsMonitor.incrementRequest();
         }
         //in distributed mode if events are duplicated in cluster, send event only if the node is receiver coordinator
-        if (!(mode == Mode.Distributed) || !isEventDuplicatedInCluster || EventReceiverServiceValueHolder.getCarbonEventReceiverManagementService().isReceiverCoordinator()) {
+        if ((!(mode == Mode.Distributed) || !isEventDuplicatedInCluster || EventReceiverServiceValueHolder.getCarbonEventReceiverManagementService().isReceiverCoordinator())
+                        && !isManagerNode) {
```

Comment:
@lafernando , all,
I have updated the pull request with the suggested optimizations. 
Thanks, Lasantha for your suggestion.